src/lava/lib/optimization/utils/generators/mis.py

```diff
@@ -138,27 +177,14 @@ def get_qubo_matrix(self, w_diag: float, w_off: float) -> np.ndarray:
             Q_ii = w_diag
             Q_ij = w_off (for i!=j) .
         """
-        return self._get_qubo_cost_from_adjacency(self._adjacency,
-                                                  w_diag,
-                                                  w_off)
-
-    def _get_random_graph_mtx(self) -> np.ndarray:
-        """
-        Creates an undirected graph with random connectivity between nodes
-        and returns its adjacency matrix.
-        """
-        np.random.seed(self.seed)
-
-        random_matrix = np.random.rand(self.num_vertices, self.num_vertices)
-        adjacency = (random_matrix < self.connection_prob).astype(int)
-
-        adjacency = np.triu(adjacency)
-        adjacency += adjacency.T - 2 * np.diag(adjacency.diagonal())
-
-        return adjacency
+        if w_off <= 2 * w_diag:
+            raise ValueError(
+                "Off-diagonal weights must be > 2 x diagonal weights."

```

Comment:
Can we also add the explanation to the message? Currently it says what must be true, but not why.